request.go

```diff
@@ -589,6 +589,12 @@ func unmarshalAttribute(
 	value = reflect.ValueOf(attribute)
 	fieldType := structField.Type
 
+	// Handle Nullable[T]
+	if strings.HasPrefix(fieldValue.Type().Name(), "Nullable[") {
```

Comment:
I'm slightly worried that the API in it's current form maybe isn't solid enough for this:

* we export `Nullable[T]` effectively telling the consumer "do with this what you will"
* this specific implementation seems like it'd be broken by a consumer wrapping nullable in their own type, right? e.g. I'm pretty sure `type nullableBool Nullable[bool]` wouldn't be unmarshaled

it'd potentially be nice to have a more rigorous, less generic API or to see if we can do more to determine the inner field type

Reply:
@brandonc had a solution for this that I forgot to add to the request side, I'll send up another commit.

Reply:
I added tests for `Nullable[bool]` that seems to be working fine.  We can get the inner type of the generic with a call to `Elem()` (thanks @bcroft), but I haven't found a way to statically check if a reflected type is some instance of `Nullable[T]`.

Reply:
Just pointing out that the reflection isn't really the issue here since the type can't be a wrapper because we depend on the specific interface of Nullable[T]--- namely, .Get()

Reply:
We could try to pin down the specific utility of the type by naming it NullableAttr, signaling that it's meant to be used with attr decorated fields.

Reply:
I updated the check for Nullable to use reflection over the available methods to approximate a type check without the string comparison.

Reply:
@notchairmk I don't think you ever have to determine the inner field type. We blindly run the underlying value through unmarshalAttribute, so whatever previous annotation/type interactions worked before will continue to work for any type T

Reply:
the main thing I was driving at was should we not export `Nullable` and instead wrap it in our explicitly supported types? if the answer is to that question is "We do want to export `Nullable`" then the helper functions (e.g. `NullableBool(v bool)`) don't really make as much sense to me. but I'm also down for whatever.

Reply:
@notchairmk I removed the helpers, I'll probably add them back in somewhere in go-tfe.